onnxruntime/core/graph/model.cc

```diff
@@ -80,6 +80,13 @@ Model::Model(const std::string& graph_name,
   }
 
   for (const auto& [domain, version] : *p_domain_to_version) {
+#ifndef ENABLE_TRAINING_CORE
+    // Do not add training opsets in a non training build.
```

Comment:
So this seems did not respect the source ModelProto, maybe it's just my personal think, If this is to just fix that set of tests we talked about, I would prefer to skip those kind of test instead of changing this way.

Reply:
When we load the source ModelProto to onnxruntime Model, we add all the opset domains listed here to the model: https://github.com/onnx/onnx/blob/be441bf70f93369d30d1e12fd97e27d2beb75b12/onnx/defs/schema.h#L965-L983.

The implication of this is that although the source model proto does not have any training onnx ops, we still add the training related onnx opset domains such as ` ai.onnx.training` or ` ai.onnx.preview.training`.

The change in this pull request will avoid adding these opset domains in a non training build. We do not remove any opset domain from the source model, just do not add the training ones in case of non training builds.

Reply:
Thanks Baiju. OK, then if p_domain_to_version points to the domain_to_version user given from Model constructor, which means user want to use a training domain op on inference build, we won't support it. But it should be fine for this rare use case.